test/tall/regression/0500/0584.unit

```diff
@@ -2,5 +2,7 @@
 bool
     listEquals<T>(List<T> a, List<T> b) => const ListEquality<T>().equals(a, b);
 <<<
-bool listEquals<T>(List<T> a, List<T> b) =>
-    const ListEquality<T>().equals(a, b);
\ No newline at end of file
+bool listEquals<T>(List<T> a, List<T> b) => const ListEquality<T>().equals(
```

Comment:
Kinda wish this split at the `=>` instead. Especially for the small `a, b` parameters.

Reply:
Yeah, me too. This is a really tricky corner that I've seen come up in a few places. There are many examples where (to me at least) it's obviously better to split the argument list:

```dart
widget = SomeWidget(
  someParam: value,
  anotherParam: anotherValue,
);
```

But there do sometimes seem to be places like this example where that looks weird and it would be better to split at the `=`/`:`/`=>`. But I've yet to intuit any sort of rule to determine when to split the arguments and when not to.

Maybe it has something to do with the size of the argument list compared to the size of the stuff before the `(`? I'm not sure.

Reply:
+1

How difficult would it be to have it prefer to split at the `=>` when the expression fits on a single line at that indentation level? Would that heuristic be enough, or would it lead to weird editing experiences?